### ISSUE TITLE
Long session name hangs zellij

### ISSUE THREAD

**Issue by Lilja** [zellij-org/zellij#2591]
`zellij --session reallyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy` works. 
`zellij --session reallyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy` is hanging the program.

The latter one is 37 characters long

**Comment by imsnif** [comment]
Nice find!

This is the error in the logs:
```
ERROR  |zellij_utils::errors::not| 2023-07-04 16:03:50.078 [server_listener] [zellij-utils/src/errors.rs:601]: Panic occured:
             thread: server_listener
             location: At zellij-server/src/lib.rs:281:25
             message: socket path should not be longer than 108 bytes
```

The fix for this should be validating `session` on the `clap` level in `zellij_utils/src/cli.rs`.

**Comment by deepsghimire** [comment]
I would like to contribute to this issue

**Comment by imsnif** [comment]
Go for it :)

**Comment by deepsghimire** [comment]
Hi! I was trying to reproduce the issue and the maximum length of session name that worked for me (no hang) was  78 characters. (need help here)
And the way I intended to validate the session is by defining  the function:

```rust
fn  validate_session_length(name: &str) -> Result<String,String> {
    if name.len() > 78 {  // or whatever length that works
        Err(format!("Session name must be less than 78 bytes"))
    } else {
        return Ok(name.to_owned())
    }
}
```

at `zellij_utils/src/cli.rs` and using this function as value_parser for `session` of the `CliArgs` struct. 

Do you know if this is the correct way to do this? Did I miss anything? Sorry for asking too many questions.

**Comment by imsnif** [comment]
@deepsghimire - what I can say is that we want the user to get a proper error in the CLI, with coloring, the correct exit status and everything. The rest is up to you as the implementer.

I think the session name limit is platform specific (for me the number was 108) - so do look into that, we might need to be a little smarter about this.

Thanks for looking into this!

**Comment by super-secret-github-user** [comment]
In my case, the issue presents itself when the session length is at least 38 characters

**Comment by super-secret-github-user** [comment]
> Nice find!
> 
> This is the error in the logs:
> 
> ```
> ERROR  |zellij_utils::errors::not| 2023-07-04 16:03:50.078 [server_listener] [zellij-utils/src/errors.rs:601]: Panic occured:
>              thread: server_listener
>              location: At zellij-server/src/lib.rs:281:25
>              message: socket path should not be longer than 108 bytes
> ```
> 
> The fix for this should be validating `session` on the `clap` level in `zellij_utils/src/cli.rs`.

How can I get zellij to produce similar logs for me? I do not get this even when running zellij with the `--debug` flag.

**Comment by deepsghimire** [comment]
@super-secret-github-user I used `cargo xtask run`

**Comment by opax** [comment]
It seems this is not really fixed; on my ARM mac with zellij 0.39.2, running

```sh
zellij -s 1234567890123456789012345678901234567
```

yields

```
ERROR  |zellij_utils::errors::not| 2024-03-04 19:03:25.984 [server_listener] [zellij-utils/src/errors.rs:632]: Panic occured:
             thread: server_listener
             location: At zellij-server/src/lib.rs:286:73
             message: called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidInput, error: "socket path should not be longer than 104 bytes" }
```
